torch/_dynamo/symbolic_convert.py

```diff
@@ -1666,8 +1666,20 @@ def __init__(
 
         vars = list(code_options["co_varnames"])
         vars.extend(x for x in self.cell_and_freevars() if x not in vars)
+
         self.symbolic_locals = collections.OrderedDict(
-            (k, VariableBuilder(self, LocalSource(k))(f_locals[k]))
+            (
+                k,
+                VariableBuilder(
+                    self,
+                    LocalInputSource(
+                        k,
+                        code_options["co_varnames"].index(k)
+                        if k in code_options["co_varnames"]
+                        else None,
```

Comment:
is it possible for this to be None? (aka to have something in `f_locals`, that doesn't have a slot in `co_varnames`)?

Reply:
this iterates over vars, gated by f_locals inclusion - I did not want to guess about their contract here.

We have an assert upstream of assert len(code_options["co_varnames"]) == code_options["co_nlocals"]  - there is also `co_freevars` with some overlap with f_locals... so I figured this was safer.